packages/pigeon/lib/kotlin_generator.dart

```diff
@@ -64,6 +65,19 @@ class KotlinOptions {
   }
 }
 
+/// Class that manages all Kotlin code generation.
+class KotlinGenerator extends Generator<KotlinOptions> {
+  /// Instantiates a Kotlin Generator.
+  KotlinGenerator();
+
+  /// Generates Kotlin files with specified [KotlinOptions]
+  @override
+  void generate(
+      KotlinOptions languageOptions, Root root, StringSink sink, FileType _) {
+    generateKotlin(languageOptions, root, sink);
```

Comment:
You might want to add an assert that the type is `source` in all the languages where that's the only expected value.

Reply:
I could do that. The parameter is never used either way though. There's no added safety of adding an assertion atm, and I don't see why that would change at all.

Reply:
I just figured it would be better to have it not fail if it wouldn't have any negative effects regardless. If you disagree that's fine though, I can change it.

Reply:
> if it wouldn't have any negative effects regardless

If a generator adapter were accidentally configured with the wrong set of file types it would do duplicate generation, which isn't harmful exactly, but would do a bunch of wasted work. Asserts would make type misconfiguration an obvious error instead of a silent bug.

But it's not a big deal either way.

Reply:
The actual list of file types doesn't force the generator to use them all. It's still called one at a time.

Reply:
> The actual list of file types doesn't force the generator to use them all. It's still called one at a time.

Isn't it driven by a generic for loop that just assumes that it should be called once with each item in the `fileTypeList` though?